lib/utmp.c

```diff
@@ -139,19 +140,26 @@ failtmp(const char *username, const struct utmpx *failent)
  *                    session
  *
  *	The utmp file is scanned for an entry with the same process ID.
- *	The line entered by the *getty / telnetd, etc. should also match
- *	the current terminal.
+ *	If several entries with matching PID found (due to previous problems
+ *	with utmp file cleanup), then the first entry with matching PID and
+ *	matching 'ut_line' (current terminal) is used or if no entry matches
+ *	'ut_line' then the first entry with matching PID is used.  If no matching
+ *	entry is found, then the entry matched current terminal is selected (for
+ *	the case if *getty has not set the correct PID).
```

Comment:
Please don't document the algorithm in a comment.  That should go in the commit message.  Source code comments are likely to be wrong/obsolete/detached from reality.